snapshotter_test.go

```diff
@@ -12,6 +12,25 @@ import (
 	"strings"
 )
 
+func tfile(t *testing.T) *os.File {
+	fname := os.DevNull
+	evar := os.Getenv("MBGL_TEST_OUT_DIR")
+	if evar != "" {
+		fmt.Println("outputing to ",evar)
+		os.MkdirAll(evar, 0600)
+
+		fname = strings.Replace(t.Name(), "/", "-", -1)
+		fname = filepath.Join(evar,  fname + ".png")
+	}
+
+	f, err := os.OpenFile(fname, os.O_WRONLY | os.O_CREATE | os.O_TRUNC, 0600)
+	if err != nil {
+		t.Fatalf("unexpected error %v", err)
+	}
+
+	return f
+}
+
```

Comment:
I don't like having test helper functions take in a testing.T. Let the test decide what it should do if the helper produces an error.